lib/fs.js

```diff
@@ -763,9 +780,19 @@ function readdirSync(path, options) {
   validatePath(path);
   const ctx = { path };
   const result = binding.readdir(pathModule.toNamespacedPath(path),
-                                 options.encoding, undefined, ctx);
+                                 options.encoding, !!options.withFileTypes,
+                                 undefined, ctx);
   handleErrorFromBinding(ctx);
-  return result;
+  if (!options.withFileTypes) {
+    return result;
+  } else {
```

Comment:
Super tiny nit: the else could be removed to reduce indentation. The same in other places.